src/hotspot/share/gc/g1/heapRegionManager.cpp

```diff
@@ -537,8 +537,8 @@ uint HeapRegionManager::find_highest_free(bool* expanded) {
   // Loop downwards from the highest region index, looking for an
   // entry which is either free or not yet committed.  If not yet
   // committed, expand at that index.
-  uint curr = reserved_length() - 1;
-  while (true) {
+  for (uint i = 0; i < reserved_length(); ++i) {
+    uint curr = reserved_length() - 1 - i;
```

Comment:
Maybe this instead?
```
for (uint curr = reserved_length(); curr-- > 0; ) {
```

Reply:
I would prefer not having side effect in the condition. At first glance, it's not obvious how many iteration the loop entails, `length` or `length - 1`?

Reply:
Avoiding side effects is normally good, but in this case I think it actually make the whole intent of the code clearer. We could add to the comment that we loop backwards through all reserved regions to make it clear what the bound is.

Reply:
Updated as suggested, since you both think it's better.